### PR TITLE
Unconditionally render non-form annotations in the annotation layer (bug 1851498)

### PR DESCRIPTION
**Pull request by calixteman**
The goal is to always have something which is focusable to let the user select
it with the keyboard.
It fixes the mentioned bug because, the annotation layer will now have a container
to attach the canvas for annotations having their own canvas.

### CODE REVIEW

**Comment by Snuffleupagus** [comment]
Given the changes made here, let's run all tests *before* I attempt to review this.

/botio test

**Comment by Snuffleupagus** [comment]
> It fixes the mentioned bug because, the annotation layer will now have a container
> to attach the canvas for annotations having their own canvas.

In the, perhaps rare, case that the Annotation is *truly empty* (and e.g. doesn't have it's own canvas) won't this mean that we'll now "render" pointless annotations (or at least "unnecessarily" add them to the DOM)?
Perhaps that's not a common occurrence nor really a problem, but being able to access (e.g. via the keyboard) completely empty annotations doesn't seem all that meaningful (at least to me).

However this might indeed be fine as-is, but given the scope of these changes I simply want to want to make sure to fully understand the changes (such that we hopefully avoid more regressions in this area).

File: src/display/annotation_layer.js [hunk 1]
```diff
@@ -2044,9 +2037,7 @@ class PopupElement {
       element.addEventListener("click", this.#boundToggle);
       element.addEventListener("mouseenter", this.#boundShow);
       element.addEventListener("mouseleave", this.#boundHide);
-      if (typeof PDFJSDev !== "undefined" && PDFJSDev.test("TESTING")) {
-        element.classList.add("popupTriggerArea");
-      }
+      element.classList.add("popupTriggerArea");
```
Comment (Snuffleupagus) [inline src/display/annotation_layer.js:2040]: If we do this unconditionally now, do we still need the following line?
https://github.com/mozilla/pdf.js/blob/08f9e48280dafb7d3a672e812afd760d59613430/src/display/annotation_layer.js#L2850

File: src/display/annotation_layer.js [hunk 2]
```diff
@@ -2806,7 +2725,9 @@ class FileAttachmentAnnotationElement extends AnnotationElement {
   #trigger = null;
 
   constructor(parameters) {
-    super(parameters, { isRenderable: true });
+    super(parameters, {
+      isRenderable: true,
+    });
```
Comment (Snuffleupagus) [inline src/display/annotation_layer.js:2730]: Nit: Looks like an unrelated change.

**Comment by calixteman** [comment]
> > It fixes the mentioned bug because, the annotation layer will now have a container
> > to attach the canvas for annotations having their own canvas.
> 
> In the, perhaps rare, case that the Annotation is _truly empty_ (and e.g. doesn't have it's own canvas) won't this mean that we'll now "render" pointless annotations (or at least "unnecessarily" add them to the DOM)? Perhaps that's not a common occurrence nor really a problem, but being able to access (e.g. via the keyboard) completely empty annotations doesn't seem all that meaningful (at least to me).
> 
> However this might indeed be fine as-is, but given the scope of these changes I simply want to want to make sure to fully understand the changes (such that we hopefully avoid more regressions in this area).

We could remove those empty annotations in the worker.
For example an empty annotation could have the following properties:
 - no associated popup
 - no data which could be rendered in a popup (i.e. content, title or rich text)
 - no appearance or with a null bounding-box (i.e. one dimensions at least is zero)
 - no tag

That said, right now the annotations (the non-widget ones) with a popup are "visible" by a screen reader (with NVDA at least) because they're clickable, but for annotations without a popup they've to be focusable and in this case we really need to have a container for them.

**Comment by Snuffleupagus** [comment]
> We could remove those empty annotations in the worker.
> For example an empty annotation could have the following properties:

Right, but it occurs to me that for the general API we probably don't want to do just "skip" any Annotations.

> but for annotations without a popup they've to be focusable and in this case we really need to have a container for them.

Sure, this all seems reasonable as-is. (I'll do the final review tomorrow, since it's getting a bit late.)

File: src/display/annotation_layer.js [hunk 3]
```diff
@@ -1926,14 +1925,8 @@ class ChoiceWidgetAnnotationElement extends WidgetAnnotationElement {
...
   constructor(parameters) {
-    const { data, elements } = parameters;
-    const isRenderable = !!(
-      data.titleObj?.str ||
-      data.contentsObj?.str ||
-      data.richText?.str
-    );
-    super(parameters, { isRenderable });
-    this.elements = elements;
+    super(parameters, { isRenderable: true });
```
Comment (Snuffleupagus) [inline src/display/annotation_layer.js:1928]: Is this actually correct, since PopupAnnotations can be created in two different ways?
Either indirectly using the `_createPopup` helper-method for certain types of Annotations, where this modified `isRenderable` property should be OK.
However in PDF documents where PopupAnnotations are explicitly specified in the /Annots-array, don't we still need to keep a "proper" `isRenderable` property here?

Basically, I'd expect something like the following to address the second case (which ought to work fine for the first case as well):
```suggestion
    super(parameters, { isRenderable: this.hasPopupData });
```

File: src/display/annotation_layer.js [hunk 4]
```diff
@@ -2847,14 +2766,10 @@ class FileAttachmentAnnotationElement extends AnnotationElement {
         }
       }
     }
-    trigger.classList.add("popupTriggerArea");
```
Comment (Snuffleupagus) [inline src/display/annotation_layer.js:2850]: Given the failure in test `issue16800-page2` I'm guessing that my suggestion to remove this line was probably wrong, sorry about that!

File: src/display/annotation_layer.js [hunk 5]
```diff
@@ -2847,15 +2770,13 @@ class FileAttachmentAnnotationElement extends AnnotationElement {
...
     this.#trigger = trigger;
 
-    if (
-      !data.popupRef &&
-      (data.titleObj?.str || data.contentsObj?.str || data.richText)
-    ) {
+    if (!data.popupRef && this.hasPopupData) {
       this._createPopup();
+    } else {
+      this.#trigger.classList.add("popupTriggerArea");
```
Comment (Snuffleupagus) [inline src/display/annotation_layer.js:2779]: Nit: How about using the local variable instead?
```suggestion
      trigger.classList.add("popupTriggerArea");
```

**Review by Snuffleupagus — APPROVED** [review]
r=me, sorry about the back-and-forth here; thank you!
